superset/common/query_context_processor.py

```diff
@@ -457,11 +458,17 @@ def get_data(self, df: pd.DataFrame) -> Union[str, List[Dict[str, Any]]]:
 
             result = None
             if self._query_context.result_format == ChartDataResultFormat.CSV:
-                result = csv.df_to_escaped_csv(
-                    df, index=include_index, **config["CSV_EXPORT"]
-                )
-            elif self._query_context.result_format == ChartDataResultFormat.XLSX:
-                result = excel.df_to_excel(df, **config["EXCEL_EXPORT"])
+                ndf = df.copy()
+                columns = list(df)
+                i = 0
+                for c in columns:
+                    coltype = None
+                    if i<len(coltypes):
+                        coltype = coltypes[i]
+                        if coltype == GenericDataType.NUMERIC:
+                            ndf[c] = pd.to_numeric(df[c], errors='ignore')
+                    i+=1
+                result = excel.df_to_excel(ndf, **config["EXCEL_EXPORT"])
```

Comment:
I may be missing something obvious here, but why are we no longer checking for `self._query_context.result_format == ChartDataResultFormat.XLSX`?

Reply:
that's an error when reporting my local tweaks to the fork, sorry for that